### PR TITLE
Create LESS breakpoint variables

### PR DESCRIPTION
**Pull request by ciar4n**
Pull Request for Issue # .

### Summary of Changes
Creates LESS breakpoint variables for media queries within the admin template.

### Testing Instructions
Code review of the template.css and template-rtl.css should be sufficient where only slightly varied media query widths should exist.

### CODE REVIEW

**Comment by C-Lodder** [comment]
Slight issue with these breakpoints. You cannot define `@md` as `768px`, then do:

`@media (min-width: @md)` **and** `@media (max-width: @md)`

Need to be changed to something like:

```
@xs: 320px;
@sm: 480px;
@md: 768px;
@lg: 979px;
@xl: 1200px;

@sm-max: @sm - 1;
@md-max: @md - 1;
@lg-max: @lg - 1;
@xl-max: @xl - 1;
```

Then when doing the media queries:

```
@media (min-width: @md-max)
```

```
@media (max-width: @md)
```

**Comment by ciar4n** [comment]
Yea that crossed my mind. My understanding is that they would simply cascade and the second rule would apply if a screen width was exactly on a breakpoint.

As we have a pretty random order of min/max width media queries, it makes more sense to do as you suggest.

I'll sort this out later.

**Comment by ciar4n** [comment]
Sorted now... thx Charlie

File: administrator/templates/isis/css/template-rtl.css [hunk 1]
```diff
@@ -5447,7 +5447,7 @@ a.badge:focus {
 		margin-left: 0;
 	}
 }
-@media (max-width: 767px) {
+@media (max-width: 768px) {
```
Comment (C-Lodder) [inline administrator/templates/isis/css/template-rtl.css]: Still not right (767px)

File: administrator/templates/isis/css/template-rtl.css [hunk 2]
```diff
@@ -5598,7 +5598,7 @@ a.badge:focus {
 		padding-right: 10px;
 	}
 }
-@media (min-width: 768px) {
+@media (min-width: 769px) {
```
Comment (C-Lodder) [inline administrator/templates/isis/css/template-rtl.css]: and here (768px)

File: administrator/templates/isis/css/template-rtl.css [hunk 3]
```diff
@@ -8658,12 +8658,12 @@ html[dir=rtl] #toolbar #toolbar-help {
 .btn-subhead {
 	display: none;
 }
-@media (min-width: 481px) {
+@media (min-width: 480px) {
```
Comment (C-Lodder) [inline administrator/templates/isis/css/template-rtl.css]: and here (479px)

**Review by C-Lodder — APPROVED** [review]

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 4ae5b18c6afdd42919670a6cbea4aca69ac36f2e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18276">issues.joomla.org/tracker/joomla-cms/18276</a>.</sub>

**Comment by csthomas** [comment]
I do not have an experience, but why?
`@lg: 979px;`
I would expect:
`@lg: 980px;`

It is related to current changes:
```diff
-@media (max-width: 979px) {
+@media (max-width: 978px) {
```

**Comment by ciar4n** [comment]
In most cases the variables match the already existing values. I believe the 979px value comes from the bootstrap original breakpoint.. https://github.com/joomla/joomla-cms/blob/staging/media/jui/less/variables.less#L179

**Comment by ghost** [comment]
@csthomas can you please test so we get the second Test?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18276">issues.joomla.org/tracker/joomla-cms/18276</a>.</sub>

**Comment by csthomas** [comment]
I did code review and I see one problem:
```diff
-@media (max-width: 979px) {
+@media (max-width: 978px) {
```

IMO this changes in a few lines is wrong and should be reverted. max-width should be 979px.
and variable `@lg` should be set to: `980px;`

I talked with Charlie about it, I think he should agree with me but he has not answered yet to my last question.
@C-Lodder Could you confirm?

**Comment by C-Lodder** [comment]
ah yes that's correct.


`@lg` should be `980px` so that we match up with the BS2 grid:


-----------------

   ```
 /* Large desktop */
    @media (min-width: 1200px) { ... }
     
    /* Portrait tablet to landscape and desktop */
    @media (min-width: 768px) and (max-width: 979px) { ... }
     
    /* Landscape phone to portrait tablet */
    @media (max-width: 767px) { ... }
     
    /* Landscape phones and down */
    @media (max-width: 480px) { ... }
```

**Review by C-Lodder — APPROVED** [review]

**Comment by csthomas** [comment]
I have tested this item :white_check_mark: successfully on 63423559c98d38ce9ca64848a25d5230e1219d4c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18276">issues.joomla.org/tracker/joomla-cms/18276</a>.</sub>

**Comment by ghost** [comment]
@Quy can you please test again?

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 63423559c98d38ce9ca64848a25d5230e1219d4c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18276">issues.joomla.org/tracker/joomla-cms/18276</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests.

**Comment by mbabker** [comment]
Apparently merging another PR caused this to conflict...

**Pull request merged**